src/parallel_progress.jl

```diff
@@ -0,0 +1,376 @@
+                     auto_reset_timer = true)
+
+allows to call the `progresses` and `mainprogress` from different workers
+ - `progresses`: contains the different progressbars
+ - `mainprogress`: main progressbar
+ - `enabled`: `enabled == false` doesn't show anything and doesn't open a channel
+ - `auto_close`: if true, the channel will close when all progresses are finished, otherwise,
+ when mainprogress finishes or with `close(p)`
+ - `count_finishes`: if false, main_progress will be the sum of the individual progressbars,
+ if true, it will be equal to the number of finished progressbars
+ - `count_overshoot`: overshooting progressmeters will be counted in the main progressmeter
+ - `auto_reset_timer`: tinit in progresses will be reset at first call
+
+use p[i] to access the i-th progressmeter, and p[0] to access the main one
+
+# Example
+```jldoctest
+julia> using Distributed
+julia> addprocs(2)
+julia> @everywhere using ProgressMeter
+julia> p = MultipleProgress(fill(10,5); desc="global ", kws=[(desc="task \$i ",) for i in 1:5])
+       pmap(1:5) do x
+           for i in 1:10
+               sleep(rand())
+               next!(p[x])
+           end
+           sleep(0.01)
+           myid()
+       end
+```
+"""
+function MultipleProgress(progresses::AbstractVector{<:AbstractProgress},
+                          mainprogress::AbstractProgress;
+                          enabled = true,
+                          auto_close = true,
+                          count_finishes = false,
+                          count_overshoot = false,
+                          auto_reset_timer = true)
+    !enabled && return MultipleProgress(FakeChannel(), length(progresses))
+
+    channel = RemoteChannel(() -> Channel{NTuple{4,Any}}(1024))
+    mp = MultipleProgress(channel, length(progresses))
+    @async runMultipleProgress(progresses, mainprogress, mp;
+        auto_close=auto_close,
+        count_finishes=count_finishes, 
+        count_overshoot=count_overshoot, 
+        auto_reset_timer=auto_reset_timer)
+    return mp
+end
+
+"""
+    MultipleProgress(progresses::AbstractVector{Progress}; count_finishes=false, kwmain=(), kw...)
+
+is equivalent to:
+
+    MultipleProgress(progresses, Progress(main_length; kwmain...); count_finishes, kw...)
+"""
+function MultipleProgress(progresses::AbstractVector{Progress}; 
+                          count_finishes=false, kwmain=(), kw...)
+    main_length = count_finishes ? length(progresses) : sum(p->p.n, progresses)
+    mainprogress = Progress(main_length; kwmain...)
+    return MultipleProgress(progresses, mainprogress; count_finishes=count_finishes, kw...)
+end
+
+"""
+    MultipleProgress(progresses::AbstractVector{<:AbstractProgress}; count_finishes=false, kwmain=(), kw...)
+
+is equivalent to:
+
+    MultipleProgress(progresses, prog; count_finishes, kw...)
+
+with `prog` a `Progress` of the same length as `progresses` if `count_finishes == true`, 
+otherwise, `prog` is a `ProgressUnknown`
```

Comment:
Couldn't this be documented in the form of a default argument `mainprogresses=Progress(...)` in the main docstring, rather than as a separate docstring?

Reply:
I'm not very good at documenting things, I've merged it in the main docstring like you suggested, but I don't know if it's very clear

```
- `mainprogress`: main progressbar, defaults to `Progress` or `ProgressUnknown`,
 according to `count_finishes` and whether all progresses have known length or not
```